src/tools/pkcs11-tool.c

```diff
@@ -6556,9 +6556,13 @@ static int read_object(CK_SESSION_HANDLE session)
 						ASN1_PRINTABLESTRING_free(curve);
 					} else if (d2i_ASN1_OBJECT(&obj, &a, (long)len) != NULL) {
 						int nid = OBJ_obj2nid(obj);
-						if (nid != NID_ED25519 && nid != NID_X25519) {
-							util_fatal("Unknown curve OID, expected NID_ED25519 (%d), got %d",
-									NID_ED25519, nid);
+						if (nid != NID_ED25519 &&
+								nid != NID_X25519 &&
+								nid != NID_ED448 &&
+								nid != NID_X448) {
+							util_fatal("Unknown curve OID, expected NID_ED25519 (%d), NID_X25519 (%d), NID_ED448 (%d), NID_X448 (%d), got %d",
+									NID_ED25519, NID_X25519,
+									NID_ED448, NID_X448, nid);
 						}
```

Comment:
This may be a problem if using libressl and they did not define all the nids. May need some ifdefs.

Note that OpenSC currently does not support any tokens that can do ED448 or X448. SO if you have one that would be a first. 

Code in pkcs11-tool for these was added but never tested.   

This looks reasonable.

Reply:
I added the missing #idef for libressl.
Let's see how it goes with the automatic checks.